### ISSUE TITLE
Destructuring syntax assignments

### ISSUE THREAD

**Issue by Marmare314** [typst/typst#696]
```
let x = 3
if x > 3 {
    let (x, y) = (1, 2)
}
x // x is still 3
```
I believe in this case the destructuring syntax should assign instead of creating a new local variable.

**Comment by SUPERCILEX** [comment]
Disagree, `let` means create a new binding. You're looking for something like `(x, let y) = (1, 2)`.

**Comment by Marmare314** [comment]
Given the current behaviour of `let` this seems like a better fit.

**Comment by astrale-sharp** [comment]
then.. is this completed? Cause I don't think it's the current behavior

**Comment by Marmare314** [comment]
Yes that is the current behaviour, see the example in my original comment.

The only thing that was not addressed is the syntax proposed by @SUPERCILEX. But this would be very hard to implement and is not supported by rust either. That is why I added #703 instead.

**Comment by astrale-sharp** [comment]
Okay! Thanks for your answer!
________________________________
